server/src/main/java/org/elasticsearch/ingest/PipelineProcessor.java

```diff
@@ -61,7 +64,7 @@ public String getType() {
         return TYPE;
     }
 
-    String getPipelineName() {
+    TemplateScript.Factory getPipelineName() {
```

Comment:
I think it would avoid a bit of confusion if the getter is renamed to `getPipelineNameTemplate` or something along those lines (it took me a while to understand we are not instantiating a name here https://github.com/elastic/elasticsearch/pull/49030/commits/91d8ada4695f8395250eaae625f8ce7dfe57492d#diff-579dffc1e22e3db13c41f685046b2891R443)